docs/components/modeler/element-templates/template-properties.md

```diff
@@ -851,6 +853,78 @@ When `zeebe:script` is used, `zeebe:taskDefinition` cannot be used on the same e
 If the input `type` is `String` or `Text`, then [`feel`](#adding-feel-editor-support-feel) must be set to `required`"
 :::
 
+### Ad-hoc sub-processes: `zeebe:adHoc`
+
+| **Binding `type`**         | `zeebe:adHoc`                                                                                                                        |
+| -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| **Valid property `type`s** | `Hidden`                                                                                                                             |
+| **Binding parameters**     | `property`: The name of the property.<br/>Supported properties: `outputCollection`, `outputElement`, and `activeElementsCollection`. |
+| **Mapping result**         | `<zeebe:adHoc [property]="[userInput]" />`                                                                                           |
+
+The `zeebe:adHoc` binding marks a sub-process as ad-hoc when deployed to Zeebe. When configured, contained activities can be executed independently without following a predefined sequence flow.
+
+### Example
+
+```json
+{
+  ...,
+  "appliesTo": [ "bpmn:AdHocSubProcess" ],
+  "elementType": { "value": "bpmn:AdHocSubProcess" },
+  "properties": [
+    ...,
+    {
+      "type": "Hidden",
+      "binding": { "type": "zeebe:adHoc", "property": "outputCollection" },
+      "value": "results"
+    },
+    {
+      "type": "Hidden",
+      "binding": { "type": "zeebe:adHoc", "property": "outputElement" },
+      "value": "={ id: results._meta.id, name: results._meta.name, content: results }"
+    }
+  ]
+}
+```
+
+### Example with `activeElementsCollection`
+
+```json
+{
+  ...,
+  "appliesTo": [ "bpmn:AdHocSubProcess" ],
+  "elementType": { "value": "bpmn:AdHocSubProcess" },
+  "properties": [
+    {
+      "type": "Hidden",
+      "binding": {
+        "type": "property",
+        "name": "cancelRemainingInstances"
+      },
+      "value": "false"
+    },
+    {
+      "type": "String",
+      "feel": "required",
+      "binding": {
+        "type": "property",
+        "name": "completionCondition"
+      }
+    },
+    {
+      "type": "Hidden",
+      "binding": { "type": "zeebe:adHoc", "property": "activeElementsCollection" },
+      "value": "=anActiveElementsCollection"
+    }
+  ]
+}
+```
+
+:::note
+The `zeebe:adHoc` binding can only be used with elements of type `bpmn:AdHocSubProcess`.
+
+The `outputCollection` property defines where ad-hoc execution results are collected, while `outputElement` specifies the structure of each result item.
```

Comment:
🔧 we could also add that rather than activeElementsCollection, one can also template a job worker as an implementation. Eg, if you look into the modeler, you can always set output collection and then you can decide between a BPMN and a job worker implementation